docs/signature/README.md

```diff
@@ -0,0 +1,228 @@
+  },
+  "layers": []
+}
+```
+
+## *Signature* Property Descriptions
+
+- **`signed`** *object*
+
+  This REQUIRED property provides the signed content.
+
+  - **`iat`** *integer*
+
+    This OPTIONAL property identities the time at which the manifests were presented to the notary. This field is based on [RFC 7519 Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6). When used, it does not imply the issue time of any signature in the `signatures` property.
+
+  - **`nbf`** *integer*
+
+    This OPTIONAL property identifies the time before which the  signed content MUST NOT be accepted for processing. This field is based on [RFC 7519 Section 4.1.5](https://tools.ietf.org/html/rfc7519#section-4.1.5).
+
+  - **`exp`** *integer*
+
+    This OPTIONAL property identifies the expiration time on or after which the signed content MUST NOT be accepted for processing. This field is based on [RFC 7519 Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4).
+
+  - **`digest`** *string*
+
+    This REQUIRED property is the *digest* of the target manifest, conforming to the requirements outlined in [Digests](https://github.com/opencontainers/image-spec/blob/master/descriptor.md#digests). If the actual content is fetched according to the *digest*, implementations MUST verify the content against the *digest*.
+
+  - **`size`** *integer*
+
+    This REQUIRED property is the *size* of the target manifest. If the actual content is fetched according the *digest*, implementations MUST verify the content against the *size*.
+
+  - **`references`** *array of strings*
+
+    This OPTIONAL property claims the manifest references of its origin. The format of the value MUST matches the [*reference* grammar](https://github.com/docker/distribution/blob/master/reference/reference.go). With used, the `x509` signatures are valid only if the domain names of all references match the Common Name (`CN`) in the `Subject` field of the certificate.
+
+- **`signature`** *string*
+
+  This REQUIRED property provides the signature of the signed content. The entire signature file is valid if any signature is valid. The `signature` object is influenced by JSON Web Signature (JWS) at [RFC 7515](https://tools.ietf.org/html/rfc7515).
+
+  - **`typ`** *string*
+
+    This REQUIRED property identifies the signature type. Implementations MUST support at least the following types
+
+    - `x509`: X.509 public key certificates. Implementations MUST verify that the certificate of the signing key has the `digitalSignature` `Key Usage` extension ([RFC 5280 Section 4.2.1.3](https://tools.ietf.org/html/rfc5280#section-4.2.1.3)).
+
+    Implementations MAY support the following types
+
+    - `tuf`: [The update framework](https://theupdateframework.io/).
+
+  - **`sig`** *string*
+
+    This REQUIRED property provides the base64-encoded signature binary of the specified signature type.
+
+  - **`alg`** *string*
```

Comment:
The problem is that key management really shouldn't be considered a completely orthogonal problem to this spec. Consider this unsigned `alg` portion of the `signature` here. If you are not careful, you are going to run into the same [problem ](https://github.com/theupdateframework/tuf/issues/356) where the signature algorithm is controlled by attackers instead, which is a problem that both py-TUF and Notary-v1 has seen before.

Reply:
Some more details on why the `alg` shouldn't be included in the unsigned  portion is available in [TAP 9](https://github.com/theupdateframework/taps/blob/master/tap9.md).

Reply:
Also, if i understanding this signature specification correctly, then they are also allowing for the None cipher due to the fact they aim for RFC compliance?

If so, this essentially opens the door for yet another arbitrary signature vulnerability in a JWT library as outlined [here](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/).

The other alternative would be to say "we support JWT-like signatures but we will throw away half of the RFC because it's a walking vulnerability"

Reply:
Please see #3